lib/dgram.js

```diff
@@ -282,38 +279,38 @@ Socket.prototype.send = function(buffer,
   if (typeof callback !== 'function')
     callback = undefined;
 
-  self._healthCheck();
+  this._healthCheck();
 
-  if (self._bindState == BIND_STATE_UNBOUND)
-    self.bind({port: 0, exclusive: true}, null);
+  if (this._bindState == BIND_STATE_UNBOUND)
+    this.bind({port: 0, exclusive: true}, null);
 
   // If the socket hasn't been bound yet, push the outbound packet onto the
   // send queue and send after binding is complete.
-  if (self._bindState != BIND_STATE_BOUND) {
+  if (this._bindState != BIND_STATE_BOUND) {
     // If the send queue hasn't been initialized yet, do it, and install an
     // event handler that flushes the send queue after binding is done.
-    if (!self._sendQueue) {
-      self._sendQueue = [];
-      self.once('listening', function() {
+    if (!this._sendQueue) {
+      this._sendQueue = [];
+      this.once('listening', () => {
         // Flush the send queue.
-        for (var i = 0; i < self._sendQueue.length; i++)
-          self.send.apply(self, self._sendQueue[i]);
-        self._sendQueue = undefined;
+        for (var i = 0; i < this._sendQueue.length; i++)
+          this.send.apply(this, this._sendQueue[i]);
```

Comment:
@targos I remember reading that the spread operator isn't too friendly to v8 optimizations and to performance in general.